src/drive/web/modules/upload/index.js

```diff
@@ -401,12 +407,21 @@ export const uploadFilesFromNative =
             contentType: file.file.type
           })
         } else {
-          await doMobileUpload(client, file.file.fileUrl, {
-            ...fileOpts,
-            contentType: file.file.type
-          })
+          if (alternateUploader) {
+            // We don't want to wait for the upload to finish, we just want to
+            // send the file to upload to the alternate uploader, it will update by itself
+            alternateUploader(client, file.filePath, {
+              ...fileOpts,
+              contentType: file.mimeType
+            })
+          } else {
+            await doMobileUpload(client, file.file.fileUrl, {
+              ...fileOpts,
+              contentType: file.file.type
+            })
+          }
         }
-        dispatch(removeFileToUploadQueue(file.file))
+        if (!alternateUploader) dispatch(removeFileToUploadQueue(file.file))
```

Comment:
Can't we get the same mechanism between alternateUploader and other?

Because : 
- We add to the UploadQueue here, no matter if this is an alternateUploader or not 
- But we remove only if this is not an alternate. This a strange. We should ave the same behavior.

Reply:
I did so because there is no real way to know when the file will need to be removed (since it comes from a post-me message). It was too complex to couple post-me with the upload module. Instead we dispatch a removeFile to the current state whenever we receive the instruction

Reply:
In that case, can the alternateUploader send by itself the add to queue?

Reply:
> In that case, can the alternateUploader send by itself the add to queue?

yes that would make things more aligned